lib/features/public_asset/presentation/public_asset_controller.dart

```diff
@@ -0,0 +1,170 @@
+import 'package:tmail_ui_user/features/identity_creator/presentation/utils/identity_creator_constants.dart';
+import 'package:tmail_ui_user/features/public_asset/domain/usecase/delete_public_assets_interactor.dart';
+import 'package:tmail_ui_user/features/public_asset/presentation/model/public_asset_arguments.dart';
+import 'package:tmail_ui_user/features/upload/domain/extensions/platform_file_extension.dart';
+import 'package:tmail_ui_user/features/upload/domain/state/attachment_upload_state.dart';
+
+typedef PublicAssetId = Id;
+
+class PublicAssetController extends BaseController {
+  PublicAssetController(
+    this._uploadAttachmentInteractor,
+    this._createPublicAssetInteractor,
+    this._deletePublicAssetsInteractor,
+    {required this.arguments}
+  );
+
+  final UploadAttachmentInteractor _uploadAttachmentInteractor;
+  final CreatePublicAssetInteractor _createPublicAssetInteractor;
+  final DeletePublicAssetsInteractor _deletePublicAssetsInteractor;
+  final PublicAssetArguments? arguments;
+
+  List<PublicAssetId> preExistingPublicAssetIds = [];
+  List<PublicAssetId> newlyPickedPublicAssetIds = [];
+  final _publicAssetStreamGroup = StreamGroup<Either<Failure, Success>>();
+  StreamSubscription<Either<Failure, Success>>? _publicAssetStreamSubscription;
+
+  StreamSubscription<Either<Failure, Success>>? get  publicAssetStreamSubscription
+    => _publicAssetStreamSubscription;
+  Session? get session => arguments?.session;
+  AccountId? get accountId => arguments?.accountId;
+  Identity? get identity => arguments?.identity;
+
+  void _handleUploadState(Either<Failure, Success> uploadState) {
+    log('PublicAssetController::handleUploadState::${uploadState.runtimeType}');
+    uploadState.fold(
+      (failure) {},
+      (success) {
+        log('PublicAssetController::handleUploadState::success::$success');
+        if (success is SuccessAttachmentUploadState) {
+          final filePath = success.fileInfo.filePath;
+          if (PlatformInfo.isMobile && filePath != null) {
+             _deleteCompressedFileOnMobile(filePath);
+          }
+
+          final blobId = success.attachment.blobId;
+          if (session == null
+            || accountId == null
+            || blobId == null) return;
+
+          consumeState(_createPublicAssetInteractor.execute(
+            session!,
+            accountId!,
+            blobId: blobId,
+            identityId: identity?.id));
+        }
+      }
+    );
+  }
+
+  Future<void> _deleteCompressedFileOnMobile(String filePath) async {
+    log('PublicAssetController::_deleteCompressedFileOnMobile: filePath: $filePath');
+    try {
+      final File file = File(filePath);
+      if (await file.exists() && file.path.contains(IdentityCreatorConstants.prefixCompressedInlineImageTemp)) {
+        await file.delete();
+      }
+    } catch (e) {
+      logError('PublicAssetController::_deleteCompressedFileOnMobile: error: $e');
+    }
```

Comment:
Should we process at the data layer?

Reply:
rather not bloating this PR even more

Reply:
No, we need to separate it. It will make the controller handle too much code.

Reply:
move to FileUtils class

Reply:
Done